feat: Support policies in folders for reusable workflows

viccuad opened the pull request:
## Description

<!-- Please provide the link to the GitHub issue you are addressing -->
Fix https://github.com/kubewarden/kubewarden-controller/issues/806

Recommend review by commit.


See also working monorepo repo at https://github.com/viccuad/rego-policies.

## Test

<!-- Please provides a short description about how to test your pullrequest -->


<!--
```shell
cp <to_package_directory>
go test
```
-->

For monorepo policies:
- Successfully failing release with tag `ControllerContainerBlockSSHPort/v0.0.3`
  because artifacthub-pkg.yml hasn't been updated:
  https://github.com/viccuad/rego-policies/actions/runs/12352787565/job/34470569919
- Successful run and release with tag `ControllerContainerBlockSSHPort/v0.0.4`:
  https://github.com/viccuad/rego-policies/actions/runs/12352849151

For Normal policies:
- Successful CI run:
  https://github.com/viccuad/disallow-service-loadbalancer-policy/actions/runs/12352648538
- Successful :latest release:
  https://github.com/viccuad/disallow-service-loadbalancer-policy/actions/runs/12352648552
- Sucessful failing release because artifacthub-pkg.yml has not been updated:
  https://github.com/viccuad/disallow-service-loadbalancer-policy/actions/runs/12352705437/job/34470302880
- Sucessful tagged release:
  https://github.com/viccuad/disallow-service-loadbalancer-policy/actions/runs/12352738499

## Additional Information

### Tradeoff

<!-- Please describe, if any, the tradeoffs that you found acceptable in this pull request -->

### Potential improvement


TODO:
- [ ] Implement normal CI per policy on PRs (it's marked as a nice-to-have).
- [ ] After approval, rebase and substitute the `DROP` commit, consuming my fork of kubewarden/github-actions for testing, and instead bump versions for release.

fabriziosestito commented on opa-installer/action.yml:
```diff
@@ -1,13 +1,13 @@
+    description: "opa release to be installed"
     required: false
-    default: v0.32.0
+    default: v0.65.0
```
wow that was an old version, maybe we should use some updatecli here

viccuad replied:
Good point, opened https://github.com/kubewarden/github-actions/issues/139.

viccuad commented on policy-release/action.yml:
```diff
@@ -14,57 +14,78 @@ inputs:
   steps:
-    -
-      name: Login to GitHub Container Registry
+    - id: calculate-version
```
> We should also use a policy version variable here to make this agnostic.

It's not as simple, as the GH release needs to match the git tag that it was triggered with, *only* when the trigger of this workflow was indeed from a tag (e.g: `v0.1.0`) instead of the "monorepo tag" (`policy/v0.1.0` edited into `0.1.0`).

Instead of again recalculating the version from the "monorepo tag", as we do in the monorepo workflow defition, here I just add a new, non-required `policy-version` input, that when present changes the behaviour.

I can take this out of the action, but that means passing that information through the reusable workflows, and potentially editing workflows of the policies I have, as that breaks backwards compatibility.

viccuad replied:
After a tag, I think you are right and the `calculate_version` step can be dropped. Let me see.

viccuad replied:
Added some comments that remove the calculate-version from the `policy-release` action, and expect this calculation to have been taken place in the workflows, by now making `policy-release.inputs.policy-version` required.
Hence, amended the workflows to always calculate-version and pass this value down the chain.

fabriziosestito commented:
@viccuad note: DCO is missing

viccuad commented:
Thanks! The DCO is missing on the `fixup!` commit, it will get removed with the rebase.

jvanz commented on .github/workflows/reusable-test-policy-rust.yml:
```diff
@@ -40,7 +40,7 @@ jobs:
         shell: bash
       - name: Check that artifacthub-pkg.yml is up-to-date
-        uses: kubewarden/github-actions/check-artifacthub@v3.3.5
+        uses: viccuad/github-actions/check-artifacthub@main
```
Just remember to update that for the kubewarden org repository

jvanz reviewed (approved):
LGTM

viccuad commented:
Rebased with a new `calculate-version` step, that now correctly sets the `version` regardless if it was run manually by passing a `policy-version`, it was triggered via git, or triggered via a branch for `:latest, :latest-feat-foo`.

Rerun the test workflow runs, which are passing, and updated their links in the PR body.

viccuad commented:
Found a minor bug when running tests from PRs in `calculate-version` step, please don't merge.

viccuad commented:
Ready for merge.

The bug happened when, on PRs (not a run from branch nor a run from a tag), the `calculate-version` step was incorrectly going through the arm of tags. For normal repos this was fine, but for monorepos, it meant that it was taking a tag of `PolicyName/v0.1.0` and generating a `version=olicyName/v0.1.0`  instead of `version=0.1.0`, which breaks `make artifacthub-pkg.yml`.

This is now fixed, see https://github.com/kubewarden/github-actions/compare/ffc0a258bac24630fede41f0f7c2de2f771eb9e1..0698fff8784455d9278bf36fc845a9f1989eb91e.

jvanz commented on policy-release/action.yml:
```diff
@@ -14,57 +14,64 @@ inputs:
-      if: ${{ startsWith(github.ref, 'refs/tags/') }}
+      if: ${{ ! startsWith(github.ref, 'refs/heads/') }}
       env:
         COSIGN_EXPERIMENTAL: 1
```
As far as I can remember, we use this env to allow keyless signing. This is not necessary anymore. Therefore, we can remove this.

viccuad commented:
Tagged in main as `v3.4.0`.
